genesyscloud/auth_division/data_source_genesyscloud_auth_division.go

```diff
@@ -21,11 +21,12 @@ func dataSourceAuthDivisionRead(ctx context.Context, d *schema.ResourceData, m i
 	// Query division by name. Retry in case search has not yet indexed the division.
 	return util.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
 		divisionId, resp, retryable, getErr := proxy.getAuthDivisionIdByName(ctx, name)
-		if getErr != nil && !retryable {
-			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Error requesting division %s | error: %s", name, getErr), resp))
-		}
-		if retryable {
```

Comment:
think this could be the problem. retryable infinite loop here. no getErr check ?